src/Data/Graph.purs

```diff
@@ -1,152 +1,107 @@
 -- | A data structure and functions for graphs
 
 module Data.Graph
-  ( Edge(..)
-  , Graph(..)
-  , SCC(..)
+  ( Graph
```

Comment:
I would like to hide the constructor here, in case we ever want to support different graph representations later. We can always export it later, which would be non-breaking, but not the reverse. 

Also, there are invariants I might like to enforce later, like edges always being between defined keys.